src/core_ocean/shared/mpas_ocn_wetting_drying.F

```diff
@@ -0,0 +1,710 @@
+      real (kind=RKIND) :: minThickness, layerThick
+      character (len=StrKIND) :: debugFilename
+      character (len=StrKIND), pointer :: config_time_integrator
+
+      !-----------------------------------------------------------------
+      !
+      ! call relevant routines for computing coefficients
+      !
+      !-----------------------------------------------------------------
+
+      call mpas_pool_get_subpool(block % structs, 'state', statePool)
+      call mpas_pool_get_subpool(block % structs, 'mesh', meshPool)
+      call mpas_pool_get_subpool(block % structs, 'tend', tendPool)
+
+      call mpas_pool_get_array(statePool, 'layerThickness', layerThicknessCur, timeLevel=1)
+      call mpas_pool_get_array(statePool, 'layerThickness', layerThicknessNew, timeLevel=2)
+      call mpas_pool_get_dimension(meshPool, 'nCellsSolve', nCellsSolve)
+      call mpas_pool_get_array(meshPool, 'maxLevelCell', maxLevelCell)
+      call mpas_pool_get_array(tendPool, 'layerThickness', layerThicknessTend)
+      call mpas_pool_get_array(statePool, 'sshSubcycle', sshSubcycleNew, 2)
+      call mpas_pool_get_array(meshPool, 'bottomDepth', bottomDepth)
+      call mpas_pool_get_config(ocnConfigs, 'config_time_integrator', config_time_integrator)
+
+      err = 0
+
+      call mpas_log_write( 'Verifying that cells are not dry...')
+
+      ! check to make sure that there is no layer that is too dry
+      minThickness = +1.0E34
+      do iCell = 1, nCellsSolve
+        do k = 1, maxLevelCell(iCell)
+          !call mpas_log_write( 'Layer thickness = $r', realArgs=(/ layerThick/))
+          ! use ssh as a proxy too for baroclinic mode
+          if (trim(config_time_integrator) == 'split_explicit') then
+            layerThick = min(layerThicknessNew(k, iCell), (sshSubcycleNew(iCell)+bottomDepth(iCell))/maxLevelCell(iCell))
+          else
+            layerThick = layerThicknessNew(k, iCell)
+          end if
+          minThickness = min(minThickness, layerThick)
+          if ( layerThick <  minHeight ) then
+            ! report error if layerThickness constraint is violated
+            if (err == 0) then
+              debugFilename = ocn_build_log_filename('mpas_ocean_layerThickness_block_stats_', block % blockID)
+              call mpas_new_unit(debugUnit)
+              open(unit=debugUnit, file=debugFilename, form='formatted', status='unknown')
+            end if
+            err = iCell
+            write(debugUnit, '(A, I5, A, I5, A, ES14.7, A)') 'ERROR: layerThickness too small at iCell=', iCell, &
+                             ', k=', k, ' with thickness of ', layerThick , '.'
+          end if
+        end do
+      end do
+
+      call mpas_log_write('Minimum thickness is $r.', realArgs=(/minThickness/))
+
+      if ( err > 0) then
+        ! end the simulation if layerThickness constraint is violated
+        call mpas_log_write( 'Done, some cells are have dried.')
+        flush(debugUnit)
+        close(debugUnit)
```

Comment:
I thought we were safe with this close but perhaps this isn't the case.  It isn't checked for in the code at present.  The debug unit lines were just to provide some additional debugging output and could be commented to see if this fixes your issue at longer times.

Reply:
Thanks, I'll try that.